Cocktails9/Cocktails9/View/DrinksGrid.swift

```diff
@@ -1,40 +1,154 @@
 import SwiftUI
 
 struct DrinksGrid: View {
-    @StateObject private var networkManager = NetworkManager.shared
+    @StateObject private var appData = AppDataManager.shared // Shared AppDataManager
+    @State private var searchQuery: String = ""  // For holding search text
+    @State private var isFiltering: Bool = false  // For handling filter state
+    @State private var isLoading: Bool = false    // To show loading indicator
+    @State private var errorMessage: String? = nil // For handling API errors
+    @State private var isSearchBarActive: Bool = false // To control visibility of the search bar
     
     let columns = [
         GridItem(.flexible()),
         GridItem(.flexible())
     ]
     
-    
+    // Function to perform search when magnifying glass is clicked
+    private func searchCocktails() {
+        guard !searchQuery.isEmpty else { return }
+        
+        isLoading = true  // Start loading
+        errorMessage = nil // Clear previous error message
+        
+        // Perform the search API call asynchronously
+        Task {
+            await NetworkManager.searchCocktails(query: searchQuery) // Fetch drinks based on query
+            isLoading = false  // Stop loading
+            if appData.drinks.isEmpty {
+                errorMessage = "No cocktails found." // If no results are found, show a message
```

Comment:
It would be better UX to have this message written on the grid. To be precise, if there are drinks - present grid, if drinks array is empty - present some EmptyView instead with text on it